spring-integration-redis/src/test/java/org/springframework/integration/redis/store/RedisMessageGroupStoreTests.java

```diff
@@ -453,6 +456,74 @@ public void testJsonSerialization() throws Exception {
 		assertEquals(errorMessage.getOriginalMessage(), ((ErrorMessage) errorMessageResult).getOriginalMessage());
 		assertEquals(errorMessage.getPayload().getMessage(),
 				((ErrorMessage) errorMessageResult).getPayload().getMessage());
+
+		Message<Foo> fooMessage = new GenericMessage<>(new Foo("foo"));
+		try {
+			store.addMessagesToGroup(1, fooMessage);
+			fail("SerializationException expected");
+		}
+		catch (Exception e) {
+			assertThat(e.getCause().getCause(), instanceOf(IllegalArgumentException.class));
+			assertThat(e.getMessage(),
+					containsString("The class with " +
+							"org.springframework.integration.redis.store.RedisMessageGroupStoreTests$Foo and name of " +
+							"org.springframework.integration.redis.store.RedisMessageGroupStoreTests$Foo " +
+							"is not in the trusted packages:"));
+		}
+
```

Comment:
Hmmm - I don't think serialization is a problem; it's deserialization that's critical.

Or, are you trying to preempt someone from storing a message that can't be deserialized?